### ISSUE TITLE
Vulnerability in dependency snakeyaml-engine v1.0

### ISSUE THREAD

**Issue by kierendavies** [hazelcast/hazelcast#17425]
Hazelcast depends on snakeyaml-engine v1.0, which has a reported denial-of-service vulnerability [CVE-2017-18640](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-18640). This can be solved by upgrading to snakeyaml-engine v2.1.

**Comment by mmedenjak** [comment]
Thank you for the report @kierendavies ! We'll look into it and see how much we need to adapt in our codebase.

**Comment by melloware** [comment]
Yep our builds just started failing because of this vulnerability.

**Comment by melloware** [comment]
OWASP dependency-check-maven plugin is reporting this as of this week blocking builds.
```
hazelcast-4.0.2.jar/META-INF/maven/com.fasterxml.jackson.core/jackson-core/pom.xml (pkg:maven/com.fasterxml.jackson.core/jackson-core@2.9.7, cpe:2.3:a:fasterxml:jackson:2.9.7:*:*:*:*:*:*:*) : CVE-2020-24616
hazelcast-4.0.2.jar/META-INF/maven/org.snakeyaml/snakeyaml-engine/pom.xml (pkg:maven/org.snakeyaml/snakeyaml-engine@1.0, cpe:2.3:a:snakeyaml_project:snakeyaml:1.0:*:*:*:*:*:*:*) : CVE-2017-18640
```

**Comment by a1dutch** [comment]
@mmedenjak can this be backported to 3.12.x?

**Comment by melloware** [comment]
@mmedenjak this looks like it was merged yesterday however if I look inside 4.1-BETA1 its still using SnakeYaml 1.0 not 2.1???

**Comment by mmedenjak** [comment]
As mentioned in other issue, 4.1-BETA-1 release was branched from the 4.1 branch and passed release verification long before this fix was merged, which is why the fix isn't in 4.1-BETA-1 but will be in 4.1-GA (or 4.1-BETA-2, should we release it).

**Comment by mmedenjak** [comment]
Hazelcast 4.0.3 was released yesterday.
